src/Workspaces/SharedUtilitiesAndExtensions/Workspace/CSharp/Utilities/NullableHelpers/NullableHelpers.cs

```diff
@@ -0,0 +1,87 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Operations;
+using Roslyn.Utilities;
+
+#nullable enable
+
+namespace Microsoft.CodeAnalysis
+{
+    internal static class NullableHelpers
+    {
+        /// <summary>
+        /// Given an encapsulating operation, returns true if the symbol passed in
+        /// is ever assigned a possibly null value as determined by nullable flow state. Returns
+        /// null if no references are found, letting the caller determine what to do with that information
+        /// </summary>
+        public static bool? IsSymbolAssignedPossiblyNullValue(SemanticModel semanticModel, IOperation? containingOperation, ISymbol? symbol)
```

Comment:
when is `containingOperation` ever null?  this is for the var/non-var case?  so this should always be in some sort of body.  so we should always have the container, right?

Reply:
Most of the ways to get an IOperation have the ability to return null. It seems more straight forward to allow this helper to take null and return null. Maybe there's annotations we could add too? I haven't looked, but we know this is not null if the return value is not null

Reply:
oh, i may be not understanding what is containingOperatino.  i though it was the operation for the 'top level member body' the varaible is declared in.  Is that not hte case?

Reply:
Maybe I should put this a different way. The caller of this method, regardless of what assumptions we can make, will always have to do a null because `SemanticModel.GetOperation` returns `IOperation?`. We can either require that null check to happen prior to calling this method, or return `null` as an indicator that the information could not be determined. Since we already have to have a `null` return value for the case where the variable is not assigned inside the scope, I thought it best to just annotate this helper with a nullable parameter. 

Even in the local variable case it may not be a member body with top level programs in c# 9

Reply:
> We can either require that null check to happen prior to calling this method

Reply:
> Even in the local variable case it may not be a member body with top level programs in c# 9

1. by 'member body' i mean "the thing taht contains the statements :)    so whatever that is in a top-level program
2. this is a perfect example of why i would *not* want this to allow null.  caller needs to considerr these cases and do a best effort to get the container node. and, if it can't, i think that would be a suitable place to NFW so we find and fix.

in other words, since we should always have a containing IOp, actually passing null through indicates an actual bug rather than a true case to consider.